### PR TITLE
build: 1ES Pipeline fixes

### PR DESCRIPTION
**Pull request by hallvictoria**
Unit tests pipelines:
- Separates each extension's unit tests into a separate pipeline

Public build:
- Adds CodeQL to public build
- Fixes fastapi directory typo

Official build:
- Publishes build pipeline artifacts (to be used when testing dev versions)

### CODE REVIEW

File: eng/templates/official/jobs/build-artifacts.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      outputParentDirectory: $(Build.ArtifactStagingDirectory)
+      outputs:
+        - output: pipelineArtifact
+          targetPath: $(Build.SourcesDirectory)
+          artifactName: "$(EXTENSION_DIRECTORY)"
+
+    steps:
+      - task: UsePythonVersion@0
+        inputs:
+          versionSpec: '3.11'
```
Comment (YunchuWang) [inline eng/templates/official/jobs/build-artifacts.yml:26]: build the extension also for other python versions, so version compaitibilities can be tested?

File: eng/ci/base-integration-tests.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+trigger: none # ensure this is not ran as a CI build
```
Comment (YunchuWang) [inline eng/ci/base-integration-tests.yml]: the pipeline name is integration tests but we run extension unit tests inside, am i missing something ?

File: eng/templates/official/jobs/base-unit-tests.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+          versionSpec: $(PYTHON_VERSION)
+      - bash: |
+          python -m pip install --upgrade pip
+          cd azurefunctions-extensions-base
+          python -m pip install -U -e .[dev]
+        displayName: 'Install dependencies'
+      - bash: |
+          python -m pytest -q --instafail --cov=. --cov-report xml --cov-branch azurefunctions-extensions-base/tests/ --ignore='azurefunctions-extensions-bindings-blob', --ignore='azurefunctions-extensions-http-fastapi'
+        env:
+          AzureWebJobsStorage: $(AzureWebJobsStorage)
```
Comment (gavin-aguiar) [inline eng/templates/official/jobs/base-unit-tests.yml]: Do we need this for unit tests?

File: eng/templates/official/jobs/base-unit-tests.yml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+      - task: UsePythonVersion@0
+        inputs:
+          versionSpec: $(PYTHON_VERSION)
+      - bash: |
+          python -m pip install --upgrade pip
+          cd azurefunctions-extensions-base
+          python -m pip install -U -e .[dev]
+        displayName: 'Install dependencies'
+      - bash: |
+          python -m pytest -q --instafail --cov=. --cov-report xml --cov-branch azurefunctions-extensions-base/tests/ --ignore='azurefunctions-extensions-bindings-blob', --ignore='azurefunctions-extensions-http-fastapi'
```
Comment (gavin-aguiar) [inline eng/templates/official/jobs/base-unit-tests.yml]: Lets remove codecov if its not being used.

Comment (gavin-aguiar) [reply eng/templates/official/jobs/base-unit-tests.yml]: Same for all pipelines

File: eng/ci/official-build.yml [hunk 5]
```diff
@@ -42,9 +42,16 @@ extends:
...
+          - template: /eng/templates/official/jobs/base-unit-tests.yml@self
+      - stage: RunBlobTests
+        dependsOn: Build
+        jobs:
+          - template: /eng/templates/official/jobs/blob-unit-tests.yml@self
+      - stage: RunFastApiTests
         dependsOn: Build
         jobs:
-          - template: /eng/templates/ci-tests.yml@self
+          - template: /eng/templates/official/jobs/fastapi-unit-tests.yml@self
```
Comment (gavin-aguiar) [inline eng/ci/official-build.yml:57]: why is official build running fastapi tests?

File: eng/templates/official/jobs/build-artifacts.yml (hunk 1 above)
Comment (hallvictoria) [reply eng/templates/official/jobs/build-artifacts.yml:26]: Discussed offline. Going to stick with the latest stable supported version. Any issues with compatibility will be caught in the worker tests.

File: eng/ci/base-integration-tests.yml (hunk 2 above)
Comment (hallvictoria) [reply eng/ci/base-integration-tests.yml]: Renamed

File: eng/templates/official/jobs/base-unit-tests.yml (hunk 3 above)
Comment (hallvictoria) [reply eng/templates/official/jobs/base-unit-tests.yml]: For blob tests, yes. Removed from the others.

File: eng/templates/official/jobs/base-unit-tests.yml (hunk 4 above)
Comment (hallvictoria) [reply eng/templates/official/jobs/base-unit-tests.yml]: Removed

File: eng/ci/official-build.yml (hunk 5 above)
Comment (hallvictoria) [reply eng/ci/official-build.yml:57]: The official build should run tests. It's one build file that builds all the extensions and runs all of the tests, but the tests are each in a separate file.

**Review by gavin-aguiar — APPROVED** [review]

**Review by YunchuWang — APPROVED** [review]

**Pull request merged**